### PR TITLE
Cleanin' 'n' testin'

### PR DESCRIPTION
**Pull request by skalnik**
make-me was a rough project to get something out the door. This PR starts cleaning it up a bit and adding tests.

### CODE REVIEW

File: server/lib/lock.rb [hunk 1]
```diff
@@ -27,15 +27,28 @@ def update_lock(data={})
...
     end
 
-private
+    private
     def lock_key(key, default=nil)
-      read_lock({}).fetch(key.to_sym, default)
+      if read_lock.nil?
+        default
+      else
+        read_lock.fetch(key.to_sym)
```
Comment (sshirokov) [inline server/lib/lock.rb]: Still need to `fetch(key.to_sym, default)` otherwise fetch can raise `KeyError`

**Comment by sshirokov** [comment]
Might as well add a script/cibuild, because I'm used to it

File: server/lib/lock.rb (hunk 1 above)
Comment (skalnik) [inline server/lib/lock.rb]: Good call, I forgot about that. Using `#[]` won't.

**Comment by skalnik** [comment]
Is there any reason to keep non-JSON encoded args passing alive, or can we kill that? /cc @sshirokov

**Comment by sshirokov** [comment]
There's not a huge reason to keep it, I only supported it for the transition.

**Pull request merged**